src/parser.cpp

```diff
@@ -397,31 +397,33 @@ void Parser::ParserImpl::loadModel(const ModelPtr &model, const std::string &inp
         addIssue(issue);
         return;
     }
-    if (!node->isCellmlElement("model")) {
+
+    if (!((!mParseFromCellml1X && node->isCellml20Element("model")) || (mParseFromCellml1X && node->isCellml1XElement("model")))) {
         auto issue = Issue::IssueImpl::create();
         if (node->name() == "model") {
             std::string nodeNamespace = node->namespaceUri();
             if (nodeNamespace.empty()) {
                 nodeNamespace = "null";
             }
-            issue->mPimpl->setDescription("Model element is in invalid namespace '" + nodeNamespace + "'. A valid CellML root node should be in namespace '" + CELLML_2_0_NS + "'.");
+            if (mParseFromCellml1X && node->isCellml20Element("model")) {
+                issue->mPimpl->setDescription("Given model is a CellML 2.0 model, use 'parseModel(...)' instead.");
```

Comment:
The first part is about the CellML document? The user has attempted to parse a CellML 2.0 model with the CellML 1X parser. The second part is a suggestion on what should be used instead.

Reply:
I wouuld expct any libCellML issue description to be directly reusable in a tool like OpenCOR, to help the user with what went wrong. Such a message would help an OpenCOR user at all.

Reply:
where does that expectation come from? is this something we are aiming to do with libCellML?

Reply:
This one...

> I would expect any libCellML issue description to be directly reusable in a tool like OpenCOR

Reply:
It does indeed come from how I would expect libCellML to behave when used by OpenCOR. Now, I appreciate that such an issue description may be useful in another use case. So, maybe it could still be generated, but if so then the issue should have a specific type, so that it can be easily discarded by those who are not interested in it.